src/extension/src/coreTemplateStudio.ts

```diff
@@ -94,144 +84,143 @@ export class CoreTemplateStudio {
     }
   }
 
-  private constructor(process: ChildProcess, port: number, url: string) {
-    this._process = process;
-    this._port = port;
-    this._url = url;
+  private constructor(processCli: ChildProcess) {
+    this._processCli = processCli;
+    this.cliEvents = new CliEventEmitter();
+    this.readStream(this._processCli);
   }
 
-  public getPort(): number {
-    return this._port;
+  // This function is a listener, in the constructor, it gets attached
+  // then it will always get triggered when a command is write to the cli and there is responses from cli, until the process gets killed
+  public async readStream(process: ChildProcess) {
+    let data = "";
+    process.stdout.on("data", (chunk) => {
+      data += chunk;
+      var responses = data.toString().split("\n");
+      for (let i = 0; i < responses.length - 1; i++) {
+        var result = JSON.parse(responses[i]);
+        this.cliEvents.emit(result['messageType'], result['content']);
+      }
+      data = responses[responses.length - 1];
+    });
+
+    process.stderr.on("data", (data) => {
+      this.cliEvents.emit("error", data.toString());
+    });
   }
 
-  public async getProjectTypes(): Promise<any> {
-    // TODO: use this in client instead of fetching directly from API server
-    const url = new URL(CONSTANTS.API.ENDPOINTS.PROJECT_TYPE, this._url);
-    return await fetch(url.href, { method: CONSTANTS.API.METHODS.GET })
-      .then((response: Response) => {
-        return response.json();
-      })
-      .catch((error: Error) => {
-        throw Error(error.toString());
+  private async awaitCliEvent(eventName: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this.cliEvents.once(eventName, (data) => {
+        this.cliEvents.removeAllListeners();
+        resolve(data);
+      });
+      this.cliEvents.once("error", (data) => {
+        this.cliEvents.removeAllListeners();
+        reject(data);
       });
+    });
   }
 
-  public async getFrameworks(projectType: string): Promise<any> {
-    const url = new URL(CONSTANTS.API.ENDPOINTS.FRAMEWORK, this._url);
-    url.searchParams.append(
-      CONSTANTS.API.QUERY_PARAMS.PROJECT_TYPE,
-      projectType
-    );
+  public async sync(payload: ICommandPayload): Promise<any> {
+    let syncCommand = `${CONSTANTS.CLI.SYNC_COMMAND_PREFIX} -p ${payload.payload.path}\n`;
```

Comment:
Given that syncCommand is not modified after initialization this could be `const`

Reply:
good catch! fixed in my PR :)